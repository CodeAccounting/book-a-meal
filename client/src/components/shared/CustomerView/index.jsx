import React from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';
import Header from '../../shared/Header';
import Footer from '../../shared/Footer';
import Preloader from '../../shared/Preloader';
import { userPropTypes } from '../../../helpers/proptypes';

/**
 * @exports
 * @function CustomerView
 * @desc Creates CustomerView Component
 * @returns {JSX} CustomerView Component
 */
const CustomerView = ({
  isFetching, children, user, logout, type
}) => {
  const mainClass = classNames({
    'content-wrapper': true
  });

  return (
    <div className="admin user">
      <Header type="customer" user={user} logout={logout} active={type} />
      <div className="content">
        <div className={mainClass} id="has-modal">
          {isFetching && <Preloader />}
          {!isFetching && children}
        </div>
      </div>
      <Footer />
    </div>
  );
};

CustomerView.propTypes = {
  ...userPropTypes,
  logout: PropTypes.func.isRequired,
  isFetching: PropTypes.bool.isRequired,
  children: PropTypes.element.isRequired,
};

export default CustomerView;
