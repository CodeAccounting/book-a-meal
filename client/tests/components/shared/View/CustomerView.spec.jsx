import React from 'react';
import CustomerView from '../../../../src/components/shared/View';
import { customer } from '../../../setup/mockData';

const props = {
  user: customer,
  logout: jest.fn(),
  type: 'menu',
  isFetching: false,
  children: <p>Hi</p>,
  showTime: true
};

describe('CustomerView', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<CustomerView {...props} />);

    expect(toJson(wrapper)).toMatchSnapshot();
    expect(wrapper.find('Preloader').length).toBeFalsy();
  });

  it('renders Preloader when fetching', () => {
    const wrapper = shallow(<CustomerView {...props} isFetching />);

    expect(toJson(wrapper)).toMatchSnapshot();
    expect(wrapper.find('Preloader').length).toEqual(1);
  });
});
