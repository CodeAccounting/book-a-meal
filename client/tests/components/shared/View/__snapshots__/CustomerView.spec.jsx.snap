// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomerView renders Preloader when fetching 1`] = `
<div
  className="admin user"
>
  <Header
    active="menu"
    dateType="menu"
    logout={[MockFunction]}
    showTime={true}
    type="customer"
    user={
      Object {
        "email": "emiola@olasanmi.com",
        "firstname": "Emiola",
        "id": "a09a5570-a3b2-4e21-94c3-5cf483dbd1ac",
        "lastname": "Olasanmi",
        "role": "customer",
      }
    }
  />
  <div
    className="content-wrapper"
    id="has-modal"
  >
    <Preloader
      type={null}
    />
  </div>
  <Connect(Modal)
    isFetching={true}
    logout={[MockFunction]}
    showTime={true}
    type="menu"
    user={
      Object {
        "email": "emiola@olasanmi.com",
        "firstname": "Emiola",
        "id": "a09a5570-a3b2-4e21-94c3-5cf483dbd1ac",
        "lastname": "Olasanmi",
        "role": "customer",
      }
    }
  >
    <p>
      Hi
    </p>
  </Connect(Modal)>
  <Footer />
</div>
`;

exports[`CustomerView renders correctly 1`] = `
<div
  className="admin user"
>
  <Header
    active="menu"
    dateType="menu"
    logout={[MockFunction]}
    showTime={true}
    type="customer"
    user={
      Object {
        "email": "emiola@olasanmi.com",
        "firstname": "Emiola",
        "id": "a09a5570-a3b2-4e21-94c3-5cf483dbd1ac",
        "lastname": "Olasanmi",
        "role": "customer",
      }
    }
  />
  <div
    className="content-wrapper"
    id="has-modal"
  >
    <p>
      Hi
    </p>
  </div>
  <Connect(Modal)
    isFetching={false}
    logout={[MockFunction]}
    showTime={true}
    type="menu"
    user={
      Object {
        "email": "emiola@olasanmi.com",
        "firstname": "Emiola",
        "id": "a09a5570-a3b2-4e21-94c3-5cf483dbd1ac",
        "lastname": "Olasanmi",
        "role": "customer",
      }
    }
  >
    <p>
      Hi
    </p>
  </Connect(Modal)>
  <Footer />
</div>
`;
